src/stats.rs

```diff
@@ -0,0 +1,160 @@
+use std::convert::TryFrom;
+use std::sync::atomic::{AtomicUsize, Ordering};
+use std::sync::Arc;
+use std::time::{Duration, Instant};
+
+use tokio::sync::Mutex;
+use tokio::time::MissedTickBehavior;
+
+use crate::shutdown::ShutdownHandle;
+
+#[derive(Clone)]
+pub struct Stats(Arc<Inner>);
+
+struct Inner {
+    shutdown_signal: ShutdownHandle,
+    /// Total numer of calls into the MSE.
+    calls_count: AtomicUsize,
+    /// Total numer of compelted calls into the MSE.
+    calls_completed_count: AtomicUsize,
+    milliseconds_waited: AtomicUsize,
+    summary: Arc<Mutex<Summary>>,
+}
+
+struct Summary {
```

Comment:
All of these values are for the last minute right?

Reply:
The values are the last second.
The summary is calculated every second, but it is only logged every minute. I am calculating it every second to keep track of e.g. the highest tps (as it would be the highest transactions per minute otherwise). Does this make sense?